app/Http/Controllers/UserController.php

```diff
@@ -2,21 +2,72 @@
 
 namespace App\Http\Controllers;
 
+use App\Http\Requests\UserPostRequest;
 use Illuminate\Http\Request;
+use App\User;
+use Image;
 
 class UserController extends Controller
 {
+    protected $user;
+
     /**
-     * Display a listing of the resource.
+     * UserController constructor.
      *
-     * @return \Illuminate\Http\Response
+     * @param User $user dependence injection
+     */
+    public function __construct(User $user)
+    {
+        $this->user = $user;
+    }
+
+    /**
+     * Get index page
+     *
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\View\View
      */
     public function index()
     {
-        if (!isset($listUsers)) {
-            $listUsers = User::all();
+        $users = $this->user->paginate(10);
+
+        return view('users.index', ['users', $users]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('users.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request request store data user
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(UserPostRequest $request)
+    {
+        $arr = $request->all();
+        if ($request->hasFile('image')) {
+            $file = $request->file('image');
+            $fileName = time() . "-" . $file->getClientOriginalName();
+            Image::make($file)->save(public_path('images/users/'. $fileName));
+            $arr['image'] = $fileName;
+        } else {
+            $arr['image'] = 'default.jpg';
+        }
```

Comment:
I think you should define common function for save image. After, you can use it everywhere you want.